app/experimenter/experiments/forms.py

```diff
@@ -63,29 +64,75 @@ class ChangeLogMixin(object):
     def __init__(self, request, *args, **kwargs):
         self.request = request
         super().__init__(*args, **kwargs)
+        self.old_status = self.instance.status
+        if self.instance.id:
+            self.old_values = ChangeLogSerializer(self.instance).data
 
     def get_changelog_message(self):
         return ""
 
     def save(self, *args, **kwargs):
+
         experiment = super().save(*args, **kwargs)
 
+        old_values = {}
+        new_values = {}
         old_status = None
 
+        self.new_data = ChangeLogSerializer(self.instance).data
         latest_change = experiment.changes.latest()
-        if latest_change:
-            old_status = latest_change.new_status
+
+        self.set_variant_change_log_values(
+            old_values, new_values, latest_change
+        )
+
+        if self.changed_data:
+            if latest_change:
+                old_status = latest_change.new_status
+
+                for field in self.changed_data:
+                    if field in self.old_values:
+                        old_values[field] = self.old_values[field]
+
+            else:
+                prev_values = {field: None for field in self.changed_data}
```

Comment:
In the event that there are no "latest changes" this would occur when the experiment is first created. want to have old_values in the changelog have values of None, while the new_values is the value that is initially set. Not entirely sure if this is necessary looking at it again...

Reply:
It makes more sense now, thnx :)